src/architecture/principles.md

```diff
@@ -0,0 +1,287 @@
+systems.
+
+### Verification Matters
+
+The requirements above state that an implementation may forgo verification when "*it is
+possible to ascertain that verification has happened elsewhere in a trusted part of the system.*"
+This is intended as a strict requirement in which implements take trust seriously. The point is
+that it's okay to not constantly spend cycles verifying hashes in an internal setup which you
+have reasons to believe is trustworthy. This is *not* a licence to trust an arbitrary data
+source just because you like them.
+
+For instance:
+
+- A JS code snippet that fetches data from an IPFS HTTP gateway without verifying it is not an
+  IPFS implementation.
+- An IPFS HTTP gateway that verifies the data that it is pulling from arbitrary IPFS nodes
+  before serving it over HTTP is an IPFS implementation.
+- That JS piece of code in the first bullet can be turned into an IPFS implementation if it
+  fetches from a :cite[trustless-gateway] and verifies what it gets.
+
+## Self-Certifying Addressability
+
+:dfn[Authority] is control over a given domain of competence and :dfn[naming authority] is
+control over what resources are called.
+
+:dfn[Addressability] is the property of a naming system such that its names are sufficient
+for an agent to interact with the resources being named.
+
+:dfn[Verifiability] is the property of a naming system such that an agent can certify that
+the mapping between a name it uses and a resource it is interacting with is correct without
+recourse to an authority other than itself and the resource.
+
+:dfn[Self-certifying addressability] is the property of a naming system such that it is both
+addressable and verifiable: any name is sufficient to interact with a resource, and its mapping
+to that resource can be certified without recourse to additional authority. Self-certifying
+addressability is a key component of a
+[self-certifying web](https://jaygraber.medium.com/web3-is-self-certifying-9dad77fd8d81)
+and it supports capture-resistance which can help mitigate against centralization.
+
+CIDs support :ref[self-certifying addressability]. With CIDs, the authority to name a resource
+resides only with that resource and derives directly from that resource's most intrinsic
+property: its content. This frees interactions with CID-named resources from the power
+relation implicit in a client-server architecture. CIDs are the trust model of IPFS.
+
+An implementation may retrieve a CID without verifying that the resource matches it, but that
+loses the resource's naming authority. Such an implementation would be comparable to an HTTP
+client looking DNS records up from a random person's resolver: it cannot guarantee that the
+addressing is authoritative. Implementers may make informed decisions as to where in their
+systems they support verification, but they should ensure that CIDs are verified whenever
+they also have access to the resource that the CID maps to.
```

Comment:
"that CIDs are verified"

Is it the other way round? Eg, making sure content is verified against the CID when you have it?

Reply:
or both, depending on execution context?